v0/destinations/yahoo_dsp/transform.js

```diff
@@ -0,0 +1,197 @@
+const sha256 = require("sha256");
+const { BASE_ENDPOINT, ENDPOINTS } = require("./config");
+const {
+  defaultRequestConfig,
+  CustomError,
+  isDefinedAndNotNullAndNotEmpty,
+  defaultPutRequestConfig,
+  removeUndefinedAndNullValues,
+  getSuccessRespEvents,
+  getErrorRespEvents
+} = require("../../util");
+
+const {
+  getAccessToken,
+  populateIncludes,
+  populateExcludes
+} = require("./util");
+
+const populateIdentifiers = (attributeArray, { Config }) => {
+  const seedList = [];
+  const { audienceType } = Config;
+  const { hashRequired } = Config;
+  let listType;
+  if (isDefinedAndNotNullAndNotEmpty(attributeArray)) {
+    // traversing through every element in the add array
+    attributeArray.forEach(element => {
+      const keys = Object.keys(element);
+      keys.forEach(key => {
+        if (key === audienceType) {
+          listType = audienceType;
+        }
+      });
+      if (!listType) {
+        throw new CustomError(
+          `Audience type ${audienceType} not provided`,
+          400
+        );
+      }
+      if (hashRequired) {
+        seedList.push(sha256(element[audienceType]));
+      } else {
+        seedList.push(element[audienceType]);
+      }
+    });
+  }
+  return seedList;
+};
+
+const responseBuilder = async (message, destination) => {
+  const { listData } = message.properties;
+  const {
+    accountId,
+    audienceId,
+    audienceType,
+    clientId,
+    clientSecret
+  } = destination.Config;
+
+  let outputPayload = {};
+  const key = "add";
+
+  const seedListRequired = ["email", "deviceId", "ipAddress"];
+
+  if (seedListRequired.includes(audienceType)) {
```

Comment:
Please add some documentation here for better understandability

Reply:
This can also be accomodate in the below switch case

Reply:
updated